### ISSUE TITLE
[BUG] skiptrigger backward bug

### ISSUE THREAD

**Issue by swuecho** [surveyjs/survey-library#2374]
### What is the current behavior?

go forward work correctly, go backward work should goto the location before skip.

As in the example, choose page4 in Q1 will jump to page4, this is expected. However, press 'previous' button, will not return to Q1.

Tested page URL:  https://plnkr.co/edit/opMaXs1GQ9R2UvHN?preview

**Comment by andrewtelnov** [comment]
@swuecho Yes, we do nothing when a user come back on a previous page. We have a global option for the complete trigger:
Survey.settings.executeCompleteTriggerOnValueChanged (default is false).
We can add a global option for skip trigger, something like:
Survey.settings.executeSkipTriggerOnValueChanged (default is false). On setting it to true, it will execute skipt triggers on value changed. Will it work?

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@swuecho I looked at the code. Skip trigger executed on value changed. If a question that used in the trigger expression changed and the trigger expression returns the true then we focus the question. Here is [the example](https://plnkr.co/edit/k3q1khE3VXyMWLVN). I do not see an issue here.

Thank you,
Andrew

**Comment by swuecho** [comment]
> @swuecho I looked at the code. Skip trigger executed on value changed. If a question that used in the trigger expression changed and the trigger expression returns the true then we focus the question. Here is [the example](https://plnkr.co/edit/k3q1khE3VXyMWLVN). I do not see an issue here.
> 
> Thank you,
> Andrew

That works fine when go forward. 

problems appears when go backward.

I change the survey to show the problem. 


https://plnkr.co/edit/DC1BhMSuLAFk66pn


```js
{
    triggers: [
        {
            type: 'skip',
            expression: "{skipToBirthDate} = 'skip'",
            gotoName: 'birthdate',
        },
    ],
    pages: [
        {
            questions: [
                {
                    name: 'skipToBirthDate',
                    type: 'radiogroup',
                    choices: [
                        'Do nothing',
                        { value: 'skip', text: 'Skip To Birth Date' },
                    ],
                },
            ],
        },
         {
            questions: [
                {
                    name: 'email',
                    type: 'text',
                    inputType: 'email',
                    title: 'Your e-mail:',
                    placeHolder: 'jon.snow@nightwatch.org',
                    validators: [
                        {
                            type: 'email',
                        },
                    ],
                },
            ],
        },
        {
            questions: [
                {
                    name: 'birthdate',
                    type: 'text',
                    inputType: 'date',
                    title: 'Your birthdate:',
                },
            ],
        },
       
    ],
};
```
a. in page 1, chose the 'skip' radio button will goto page 2, birthdate question. no problem here.
b. click 'previous' in page 3, will go to P2. ( I think goto page 1 is more reasonable)

**Comment by swuecho** [comment]
take the example in skip trigger
```js
{
pages: [
  { name: "page5",
   elements: [
   {name: "skip4Pages", type: "radiogroup" title: "Do you want to skip the next questions about ...?", choices: ["Yes", "No"]}]
  }
]
....
{
  name: "page10",
 elements: [{name: "firstQuestionAfterSkip", type: "text"}]
}
],
triggers: [{type: "skip", expression: "{skip4Pages} = 'Yes'", gotoName: "firstQuestionAfterSkip"}]
}
```

**Comment by andrewtelnov** [comment]
> My problem is when I want to go back. For complete trigger, go back is not an option. but for skip trigger, make sense to be able to go back. otherwise, should disable the 'previous' button once in the page of gotoName.

Sorry, what do you mean? Do you want to disable the "Previous" button? We have this option, ``survey.showPrevButton = false;``

Thank you,
Andrew

**Comment by swuecho** [comment]
> > My problem is when I want to go back. For complete trigger, go back is not an option. but for skip trigger, make sense to be able to go back. otherwise, should disable the 'previous' button once in the page of gotoName.
> 
> Sorry, what do you mean? Do you want to disable the "Previous" button? We have this option, `survey.showPrevButton = false;`
> 
> Thank you,
> Andrew

I guess skip trigger by design does not support 'go back'?

**Comment by andrewtelnov** [comment]
@swuecho Now I am lost. My original [plunker example](https://plnkr.co/edit/k3q1khE3VXyMWLVN?preview) does what you need. You go to the second page, choose skip to birthdate, skip trigger runs and you are back to page1.
You have modified my plunker and put birthdate question on the last page and skipToBirthDate into the first page. It works as well. Sorry, I do not see the point. What is wrong? :(

Thank you,
Andrew

**Comment by swuecho** [comment]
> page1.

sorry for the confusion. In the scenario of your example (skip to some page before current one), it works.  but not in the scenario I want to use (skip to some page after current page). probably easier for me to use visibleIf.

If you have time, please try my original example

 https://plnkr.co/edit/opMaXs1GQ9R2UvHN?preview

![image](https://user-images.githubusercontent.com/666683/92608990-9b4bec00-f2e8-11ea-92c4-63c4c1d7c38e.png)


1. choose page4 in Q1. will jump to page4, this is expected. 

![image](https://user-images.githubusercontent.com/666683/92609713-79069e00-f2e9-11ea-9e42-4fcd9cee7feb.png)

2. press 'previous' button in page 4, will not return to Q1. 

![image](https://user-images.githubusercontent.com/666683/92609586-53799480-f2e9-11ea-9e6d-527f6050ce64.png)

let's say you read a book. You are in page 1 and skip 2..9 to page 10. now you are in page 10.  If some one ask you to go back to 'previous' page.  the 'previous' page is 1 or 9?

**Comment by andrewtelnov** [comment]
@swuecho I guess I got you now. 

> You are in page 1 and skip 2..9 to page 10. now you are in page 10. If some one ask you to go back to 'previous' page. the 'previous' page is 1 or 9?

I would expect to go to 9 from page 10. This is how kindle works, for example. Because previous page from page 10 is 9. 

Anyway, we are working on a new navigation. Take a look at [this example](https://surveyjs.io/Examples/Library?id=real-covid-19). We will have a built-in functionality like this. You will be able to comeback to any previous page, but you can't go to a next page if there is unfilled requried questions, or there are any errors, in the current page or any page that is before the page you want to go.
I believe this navigation type will solve your issue.

Thank you,
Andrew
